cfn-resources/teams/cmd/resource/resource.go

```diff
@@ -191,94 +195,96 @@ func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		currentModel.Profile = aws.String(profile.DefaultProfile)
 	}
 
-	client, handlerError := util.NewMongoDBClient(req, currentModel.Profile)
+	client, handlerError := util.NewAtlasClient(&req, currentModel.Profile)
+	atlasV2 := client.AtlasV2
 	if handlerError != nil {
 		_, _ = logger.Warnf("CreateMongoDBClient error: %v", handlerError)
 		return *handlerError, errors.New(handlerError.Message)
 	}
 
-	if !isExist(client, currentModel) {
+	team, res, err := getTeam(atlasV2, currentModel)
+	if err != nil && res != nil {
+		_, _ = logger.Debugf("error getting Team information: %s", err)
+		return progressevents.GetFailedEventByResponse(err.Error(), res), nil
+	} else if err != nil {
 		_, _ = logger.Debugf("error getting Team information: %s", *currentModel.TeamId)
 		return handler.ProgressEvent{
 			OperationStatus:  handler.Failed,
 			Message:          "Resource Not Found",
 			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
 	}
 
-	// API call
-	team, res, err := client.Teams.Get(context.Background(), *currentModel.OrgId, *currentModel.TeamId)
-	if err != nil {
-		_, _ = logger.Debugf("error getting Team information: %s", err)
-		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
-	}
-
 	teamID := cast.ToString(currentModel.TeamId)
 	orgID := cast.ToString(currentModel.OrgId)
 	teamName := cast.ToString(currentModel.Name)
 	projectID := cast.ToString(currentModel.ProjectId)
 
 	// add existing team or newly created team to project if project id exist in the request
 	if projectID != "" && len(currentModel.RoleNames) > 0 {
-		createRequest := []*mongodbatlas.ProjectTeam{{
-			TeamID:    teamID,
+		createRequest := []admin.TeamRole{{
+			TeamId:    &teamID,
 			RoleNames: currentModel.RoleNames,
 		}}
-		_, _, err := client.Projects.AddTeamsToProject(context.Background(), projectID, createRequest)
+		_, _, err := atlasV2.TeamsApi.AddAllTeamsToProject(context.Background(), projectID, &createRequest).Execute()
 		if err != nil {
 			_, _ = logger.Warnf("error adding Team(%s) to project(%s): reason : %v", teamID, projectID, err)
 		}
 	}
 
 	// rename the team
-	if team.Name != teamName {
-		_, _, err := client.Teams.Rename(context.Background(), orgID, teamID, teamName)
+	if !util.AreStringPtrEqual(team.Name, &teamName) {
+		_, _, err := atlasV2.TeamsApi.RenameTeam(context.Background(), orgID, teamID, &admin.Team{
+			Name: teamName,
+		}).Execute()
 		if err != nil {
 			_, _ = logger.Warnf("error updating Team information: %v", err)
 		}
 	}
 
 	// add/remove user to/from teams
 	if currentModel.Usernames != nil {
-		// get the current  users list for the team
-		users, _, err := client.Teams.GetTeamUsersAssigned(context.Background(), orgID, teamID)
+		// get the current users list for the team
+		paginatedResp, _, err := atlasV2.TeamsApi.ListTeamUsers(context.Background(), orgID, teamID).Execute()
```

Comment:
do we need to make multiple calls if multiple pages?

Reply:
This is a fair point, I would create a ticket to see what options we have. I think this issue of fetching a single page is present in various CFN and terraform resources.